File: scrape_m2web.py
from selenium import webdriver
from dotenv import dotenv_values
from lxml.html.soupparser import fromstring

temp = dotenv_values(".env")

options = webdriver.ChromeOptions()
options.add_argument("--headless")
options.add_argument("--disable-extensions")
options.add_argument("--disable-gpu")
options.add_argument("--disable-dev-shm-usage")
options.add_argument("--no-sandbox")
prefs = {"profile.managed_default_content_settings.images": 2, "network.cookie.cookieBehavior": 2}
options.add_experimental_option("prefs", prefs)
options.add_experimental_option('excludeSwitches', ['enable-logging'])
driver = webdriver.Chrome(executable_path=temp["DRIVER_PATH"], chrome_options=options)

driver.get("https://us2.m2web.talk2m.com/valleycarriers/Gorman%20Bros/usr/viewon/Overview.shtm")

driver.find_element_by_xpath("//*[@id='username']").send_keys("operator")
driver.find_element_by_xpath("//*[@id='password']").send_keys("operator123")
driver.save_screenshot("s1.png")
driver.find_element_by_xpath("//*[@id='connect']").click()


driver.save_screenshot("ss.png")

File: scrape_m2web.go
package main

import (
	"fmt"
	"log"
	"os"

	"github.com/tebeka/selenium"
)

// This example shows how to navigate to a http://play.golang.org page, input a
// short program, run it, and inspect its output.
//
// If you want to actually run this example:
//
//   1. Ensure the file paths at the top of the function are correct.
//   2. Remove the word "Example" from the comment at the bottom of the
//      function.
//   3. Run:
//      go test -test.run=Example$ github.com/tebeka/selenium
func main() {
	// Start a Selenium WebDriver server instance (if one is not already
	// running).

	const (
		// These paths will be different on your system.
		seleniumPath    = "vendor/selenium-server-standalone-3.4.jar"
		geckoDriverPath = "vendor/geckodriver-v0.18.0-linux64"
		port            = 8080
	)
	opts := []selenium.ServiceOption{
		selenium.StartFrameBuffer(),           // Start an X frame buffer for the browser to run in.
		selenium.GeckoDriver(geckoDriverPath), // Specify the path to GeckoDriver in order to use Firefox.
		selenium.Output(os.Stderr),            // Output debug information to STDERR.
	}
	selenium.SetDebug(true)
	service, err := selenium.NewSeleniumService(seleniumPath, port, opts...)
	if err != nil {
		panic(err) // panic is used only as an example and is not otherwise recommended.
	}
	defer service.Stop()

	// Connect to the WebDriver instance running locally.
	caps := selenium.Capabilities{"browserName": "firefox", "headless": true}
	wd, err := selenium.NewRemote(caps, fmt.Sprintf("http://localhost:%d/wd/hub", port))
	if err != nil {
		panic(err)
	}
	defer wd.Quit()

	// Navigate to the simple playground interface.
	if err := wd.Get("https://operator:operator123@us2.m2web.talk2m.com/valleycarriers/Gorman%20Bros/usr/viewon/Overview.shtm"); err != nil {
		panic(err)
	}
	file, err := os.Open("sc.png") // For read access.
	if err != nil {
		log.Fatal(err)
	}
	data, err := wd.Screenshot()

	_, err = file.Write(data)
	if err != nil {
		log.Fatal(err)
	}

}
